parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -147,9 +158,40 @@ public static Metrics footerMetrics(ParquetMetadata metadata, Stream<FieldMetric
       upperBounds.remove(fieldId);
     }
 
+    // populate upper and lower bounds for floating columns
+    updateFloatingColumnsBounds(fieldMetricsMap, metricsConfig, fileSchema, lowerBounds, upperBounds);
+
     return new Metrics(rowCount, columnSizes, valueCounts, nullValueCounts,
-        MetricsUtil.createNanValueCounts(fieldMetrics, metricsConfig, fileSchema),
-        toBufferMap(fileSchema, lowerBounds), toBufferMap(fileSchema, upperBounds));
+        MetricsUtil.createNanValueCounts(fieldMetricsMap.values().stream(), metricsConfig, fileSchema),
+        toBufferMap(fileSchema, lowerBounds),
+        toBufferMap(fileSchema, upperBounds));
+  }
+
+  private static void updateFloatingColumnsBounds(
```

Comment:
Minor: You might consider renaming this so that it doesn't assume that Iceberg is maintaining bounds only for floating point columns. Maybe `updateFromFieldMetrics`?